### PR TITLE
GraphQL Tags - permissions and mutations

### PR DESCRIPTION
**Pull request by pylipp**
This PR implements all tests from section 4.1 (tag queries) and tests 4.2.1-8.

Any authz regarding querying/creating/updating tags is implemented (i.e. checking for permission in general and, if applicable, for base-specific permission).

@flisowna @spaudanjo this PR can also serve as baseline for implementing more tags-related and permission-related functionality and tests.

### CODE REVIEW

**Review by flisowna — APPROVED** [review]
👍🏼

**Comment by pylipp** [comment]
[Update GraphQL API for new Tags structure](https://trello.com/c/SNpRIZWQ/888-update-graphql-api-for-new-tags-structure)

**Comment by pylipp** [comment]
I won't merge this until #333 is merged (for the sake of not introducing conflicts there).

**Review by spaudanjo — APPROVED** [review]

**Pull request merged**
